tron/config/config_parse.py

```diff
@@ -270,6 +272,61 @@ class ValidateSecretSource(Validator):
 valid_secret_source = ValidateSecretSource()
 
 
+def valid_permission_mode(value, config_context):
+    try:
+        decimal_value = int(
+            str(value), base=8
+        )  # take in permission mode as string or int representation of an octal number. Goes from 0 to 4095 in decimal.
+    except ValueError:
+        error_msg = "Could not parse %s as octal permission mode at %s"
+        raise ConfigError(error_msg % (value, config_context.path))
+    if decimal_value > 4095 or decimal_value < 0:
+        error_msg = "Octal permission mode %s out of bound at %s"
+        raise ConfigError(error_msg % (value, config_context.path))
+    return str(value)
+
+
+class ValidateSecretVolumeItem(Validator):
+    config_class = ConfigSecretVolumeItem
+
+    validators = {
+        "key": valid_string,  # name of current secret
+        "path": valid_string,  # New secret filename
+        "mode": valid_permission_mode,  # Octal permission mode
+    }
+
+
+valid_secret_volume_item = ValidateSecretVolumeItem()
+
+
+class ValidateSecretVolume(Validator):
+    config_class = ConfigSecretVolume
+
+    optional = True
+    defaults = {
+        "default_mode": None,
+        "items": None,
+    }
+
+    validators = {
+        "container_path": valid_string,
+        "secret_volume_name": valid_string,
+        "secret_name": valid_string,
+        "default_mode": valid_permission_mode,
+        "items": build_list_of_type_validator(valid_secret_volume_item, allow_empty=True),
+    }
+
+    def post_validation(self, valid_input, config_context):
+        # check if each key of items matches the secret_name, and propagate default_mode.
```

Comment:
how are we propagating default mode?

Reply:
might also be good to leave a comment here about why we're checking that key == secret name (our secret system creates single-key k8s Secrets with the key being the secret name)

Reply:
Added propagation.